### PR TITLE
Delegate implemented

### PR DESCRIPTION
**Pull request by boriskezikov**

### CODE REVIEW

File: Swiift-project1/Controllers/SettingsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        }
+    }
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        self.definesPresentationContext = true
+    }
+}
+
+protocol SettingsVCDelegate: class  {
+    func callBack(left:String, right:String)
```
Comment (a-mkrv) [inline Swiift-project1/Controllers/SettingsViewController.swift:38]: what is callBack?
ok, I understand what you mean, but  please use more clear and talking naming

File: Swiift-project1/Services/Service.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+class Service {
+    
+    public static func generateRandomValue(leftBorder: Int, rightBorder:Int) -> Int {
+        let tmp = Int.random(in: leftBorder..<rightBorder)
+        print(tmp)
+        return tmp
+    }
+    public static func validateInput(value: String) -> Bool {
+        if (value.elementsEqual("")){
```
Comment (a-mkrv) [inline Swiift-project1/Services/Service.swift:22]: use the optional type in arguments 
checking for an empty line here and passing an empty line from `onClick` - a crutch

```
guard let value = value else {
    return
}
```

File: Swiift-project1/Services/Service.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        }
+          let trimmedString = value.trimmingCharacters(in: .whitespaces)
+          let validated = NSPredicate(format: "SELF MATCHES %@", Constants.validationRegex)
+          return validated.evaluate(with: trimmedString)
+      }
+    
+    public static func toast(title: String, message: String, vc:UIViewController) -> Void {
+        let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        vc.present(alert, animated: true, completion: nil)
+        Timer.scheduledTimer(withTimeInterval: 2.0, repeats: false, block: { _ in alert.dismiss(animated: true, completion: nil) })
```
Comment (a-mkrv) [inline Swiift-project1/Services/Service.swift:33]: block always on a new line

File: Swiift-project1/Services/Service.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+//  Created by Kezikov Boris  on 07.03.2020.
+//  Copyright © 2020 epam. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+
+
+class Service {
```
Comment (a-mkrv) [inline Swiift-project1/Services/Service.swift:14]: naming
what kind of service?

File: Swiift-project1/Constants.swift [hunk 5]
```diff
@@ -17,5 +17,9 @@ struct Constants{
     static let validationRegex: String = "^[0-9]*$"
```
Comment (a-mkrv) [inline Swiift-project1/Constants.swift:17]: global file with all constants - bad practice
its better to create separate private enum for each ViewController/Service/Models...

File: Swiift-project1/Controllers/SettingsViewController.swift [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    @IBOutlet weak var leftBorder: UITextField!
+    
+    weak var delegate: SettingsVCDelegate?
+    
+    @IBOutlet weak var but: UIButton!
+    
+    @IBAction func onClick(_ sender: UIButton) {
+        let leftText:String = leftBorder.text ?? "0"
+        let rightText:String = rightBorder.text ?? "0"
+        if (Service.validateInput(value: leftText) && Service.validateInput(value: rightText) && Int(rightText)! > Int(leftText)!){
```
Comment (a-mkrv) [inline Swiift-project1/Controllers/SettingsViewController.swift:23]: force unwrap

File: Swiift-project1/Controllers/ViewController.swift [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        self.triesCounter.text = NSLocalizedString(Constants.tries,comment: "") + "0"
+        instantiateSwipes()
+        Service.toast(title: NSLocalizedString(Constants.welcomeTitle,comment: ""),
+                      message: NSLocalizedString(Constants.welcomeMessage, comment: ""), vc: self)
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        if (checkParams()){
+            self.magicNumber = Service.generateRandomValue(leftBorder: leftBorder, rightBorder: rightBorder)
```
Comment (a-mkrv) [inline Swiift-project1/Controllers/ViewController.swift:37]: no need `self`

File: Swiift-project1/Controllers/ViewController.swift [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+            dest.delegate = self
+        default:
+            break
+        }
+    }
+    
+    @IBAction func readValue(_ sender: Any) {
+        if Service.validateInput(value: textField.text!) {
+            if let text = textField.text, let value = Int(text) {
+                ViewController.counter += 1
```
Comment (a-mkrv) [inline Swiift-project1/Controllers/ViewController.swift:57]: static variable? why?

File: Swiift-project1/Controllers/ViewController.swift [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+                
+                self.triesCounter.text = NSLocalizedString(Constants.tries,comment: "") + String(ViewController.counter)
+                
+                if value > magicNumber {
+                    
+                    Service.toast(title: NSLocalizedString(Constants.wrongTitle, comment: ""),
```
Comment (a-mkrv) [inline Swiift-project1/Controllers/ViewController.swift:63]: create separate extension file for NSLocalizedString
and use like `Constants.wrongTitle.localized`

File: Swiift-project1/Controllers/ViewController.swift [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+                } else if value < magicNumber {
+                    Service.toast(title: NSLocalizedString(Constants.wrongTitle, comment: ""),
+                                  message: NSLocalizedString(Constants.biggerMessage, comment: ""), vc: self)
+                } else {
+                    Service.toast(title: NSLocalizedString(Constants.successTitle,comment: ""),
+                                  message: NSLocalizedString(Constants.successMessage, comment: "") + String(magicNumber),vc: self)
+                    magicNumber = Service.generateRandomValue(leftBorder: leftBorder, rightBorder: rightBorder)
+                    ViewController.counter = 0
+                    gamesPlayedCounter += 1
+                    self.triesCounter.text = NSLocalizedString(Constants.tries,comment: "") + String(ViewController.counter)
```
Comment (a-mkrv) [inline Swiift-project1/Controllers/ViewController.swift:74]: no need `self`

File: Swiift-project1/Controllers/ViewController.swift [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+    }
+    
+    private func instantiateSwipes(){
+        let swipeRight = UISwipeGestureRecognizer(target: self, action: #selector(handleSwipes(_:)))
+        let swipeLeft = UISwipeGestureRecognizer(target: self, action: #selector(handleSwipes(_:)))
+        
+        swipeRight.direction = .right
+        swipeLeft.direction = .left
+        
+        self.view.addGestureRecognizer(swipeRight)
```
Comment (a-mkrv) [inline Swiift-project1/Controllers/ViewController.swift:105]: no need `self`

File: Swiift-project1/Constants.swift (hunk 5 above)
Comment (boriskezikov) [reply Swiift-project1/Constants.swift:17]: isnt better for project of this size to store consts just in one file ?

**Pull request merged**